src/rbd.cc

```diff
@@ -3158,6 +3158,34 @@ int main(int argc, const char **argv)
 	cerr << "rbd: order must be between 12 (4 KB) and 25 (32 MB)" << std::endl;
 	return EXIT_FAILURE;
       }
+      cerr << "rbd: --order is deprecated, please use --object-size" << std::endl;
+    } else if (ceph_argparse_witharg(args, i, &val, err, "--object-size", (char*)NULL)) {
+      if (!err.str().empty()) {
+	cerr << "rbd: " << err.str() << std::endl;
+	return EXIT_FAILURE;
+      }
+      int object_size = strict_sistrtoll(val.c_str(), &parse_err);
+      if (!parse_err.empty()) {
+        cerr << "rbd: error parsing --object-size " << parse_err << std::endl;
+        return EXIT_FAILURE;
+      }
+      // rbd: order must be between 12 (4 KB) and 25 (32 MB)
+      for (int i = 1; i < 27; i++) {
```

Comment:
@dillaman with this code - 

while (object_size > 0) {
  object_size = object_size >> 1;
  ++order;
}

gcc shift.c
[vumrao@fc22 Test]$ ./a.out 
Please give Object Size : 4194304 <--- 4 MB

 object_size is = 4194304
 order is = 23

[vumrao@fc22 Test]$ ./a.out 
Please give Object Size : 7340032  <---------- 7 MB

 object_size is = 7340032
 order is = 23

it would always give ceiling value as a nearest power of 2.  
For example if I will give 4 MB then also it is giving order as 23 which will make it 8MB it should have been 22. As you can see in above given run it is same for 7 MB and 4 MB.

+            // deciding floor and ceiling value for order
+           // for making object size to the nearest power of 2.
+           if ((object_size - (1 << (i-1))) < ((1 << i) - object_size)) {
+             order = i-1;
+             break;
+           }
+           order = i;
+           break;
+         } 
+      }

[vumrao@fc22 Test]$ gcc power2.c
[vumrao@fc22 Test]$ ./a.out 
Please give Object Size : 6291456 <----------- 6 MB

 object_size is = 6291456
Power is 23
[vumrao@fc22 Test]$ ./a.out 
Please give Object Size : 5242880 <------ 5 MB

 object_size is = 5242880
Power is 22
[vumrao@fc22 Test]$ ./a.out 
Please give Object Size : 4194305 <----- 4 MB
 
 object_size is = 4194305
Power is 22
[vumrao@fc22 Test]$ 

With this I am deciding if the object size is less than to second half of the range (for example as given above run - 4 MB to 8 MB , for 4, 5 MB it would be floor value which would be 4 MB and order will be 22 ) then take floor otherwise if the value is greater than or equal to the first half of the range (for example as given above run - 4 MB to 8 MB , for 6, 7 and 8 MB it would be ceiling value which would be 8 MB and order will be 23).

Please let me know your inputs.

Regards,
Vikhyat